manifests/user.pp

```diff
@@ -142,40 +142,40 @@
 #
-define accounts::user(
-  Pattern[/^present$|^absent$/] $ensure                     = 'present',
-  Pattern[/^\//] $shell                                     = '/bin/bash',
-  String $comment                                           = $name,
-  Optional[Pattern[/^\/$|^\/.*[^\/]$/]] $home               = undef,
-  Optional[String] $home_mode                               = undef,
-  Optional[Pattern[/^\d+$/]] $uid                           = undef,
-  Optional[Pattern[/^\d+$/]] $gid                           = undef,
-  Boolean $allowdupe                                        = false,
-  String $group                                             = $name,
-  Array[String] $groups                                     = [ ],
-  Boolean $create_group                                     = true,
-  Pattern[/^inclusive$|^minimum$/] $membership              = 'minimum',
-  Optional[Boolean] $forcelocal                             = undef,
-  String $password                                          = '!!',
-  Optional[String] $salt                                    = undef,
-  Optional[Integer] $iterations                             = undef,
-  Boolean $locked                                           = false,
-  Array[String] $sshkeys                                    = [],
-  Boolean $purge_sshkeys                                    = false,
-  String $sshkey_owner                                      = $name,
-  Boolean $managehome                                       = true,
-  Boolean $managevim                                        = true,
-  Optional[String] $bashrc_content                          = undef,
-  Optional[String] $bashrc_source                           = undef,
-  Optional[String] $bash_profile_content                    = undef,
-  Optional[String] $bash_profile_source                     = undef,
-  Boolean $system                                           = false,
-  Boolean $ignore_password_if_empty                         = false,
-  Optional[String] $forward_content                         = undef,
-  Optional[String] $forward_source                          = undef,
-  Optional[Pattern[/^absent$|^\d{4}-\d{2}-\d{2}$/]] $expiry = undef,
-  Optional[String] $sshkey_custom_path                      = undef,
-  Boolean $purge_user_home  = false,
+define accounts::user (
+  Accounts::User::Allowdupe                $allowdupe                = false,
+  Accounts::User::Bash_profile_content     $bash_profile_content     = undef,
+  Accounts::User::Bash_profile_source      $bash_profile_source      = undef,
+  Accounts::User::Bashrc_content           $bashrc_content           = undef,
+  Accounts::User::Bashrc_source            $bashrc_source            = undef,
+  Accounts::User::Comment                  $comment                  = $name,
+  Accounts::User::Create_group             $create_group             = true,
+  Accounts::User::Ensure                   $ensure                   = 'present',
```

Comment:
The `ensure` parameter is usually the first.
https://puppet.com/docs/puppet/6.3/style_guide.html#attribute-ordering

Reply:
Thanks for the review, [@smortex](https://github.com/smortex)

Updated per your recommendations; also updated docs to match.